src/Microsoft.OData.Core/Json/JsonReader.cs

```diff
@@ -2650,6 +2683,244 @@ private static bool IsNumberChar(char ch)
                 ch == '-' || ch == '+';
         }
 
+        /// <summary>
+        /// Returns an interned string if the input matches a common value or is short; otherwise, returns a new string
+        /// instance.
+        /// </summary>
+        /// <remarks>This method optimizes memory usage by interning strings that are either commonly used
+        /// or short in length. For longer or unique strings, it returns a new string instance to avoid unnecessary
+        /// interning overhead.</remarks>
+        /// <param name="span">A read-only span of characters representing the input string to process.</param>
+        /// <returns>An interned string if the input matches a predefined common value or if its length is 10 characters or
+        /// fewer; otherwise, a new string instance representing the input.</returns>
+        private static string InternIfCommon(ReadOnlySpan<char> span)
+        {
+            if (span.IsEmpty)
+            {
+                return string.Empty;
+            }
+
+            if (span.Length == 1)
+            {
+                char c = span[0];
+                if ((c >= '0' && c <= '9') || (c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z'))
+                {
+                    return c.ToString();
+                }
+
+                if (c == '-') return "-";
+                if (c == '_') return "_";
+                if (c == '.') return ".";
+                if (c == ',') return ",";
+                if (c == ':') return ":";
+            }
+
+            // For known property names, return static interned instances
+            if (TryGetMatchingCommonValueString(span, out string commonValue))
+            {
+                return commonValue;
+            }
+
+            // For strings up to length 10, intern if they start with '@' (common for OData annotations)
+            if (span.Length <= 10 && span[0] == '@')
+            {
+                // These are common starting characters for OData properties/values
+                return string.Intern(span.ToString());

```

Comment:
How much does string interning contribute to the performance gains reported in your benchmarks?